proctl/proctl_darwin.go

```diff
@@ -172,14 +172,14 @@ func (dbp *DebuggedProcess) findExecutable() (*macho.File, error) {
 	return macho.Open(C.GoString(pathptr))
 }
 
-func trapWait(dbp *DebuggedProcess, pid int) (int, *sys.WaitStatus, error) {
+func trapWait(dbp *DebuggedProcess, pid int) (int, error) {
 	port := C.mach_port_wait(dbp.os.exceptionPort)
 	if port == 0 {
-		return -1, nil, ProcessExitedError{}
+		return -1, ProcessExitedError{Pid: pid}
```

Comment:
This should be `dbp.Pid`, since we're waiting on a mach port here, the pid that gets passed in means nothing to us, it's only for parity with the Linux side.